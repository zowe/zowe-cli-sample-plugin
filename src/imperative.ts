/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import {IImperativeConfig} from "@brightside/imperative";

const config: IImperativeConfig = {
    commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
    overrides: {
        CredentialManager: "./overrides/CredentialManager.override"
    },
    pluginHealthCheck: __dirname + "/healthCheck.Handler",
    pluginSummary: "Zowe CLI sample plug-in",
    pluginAliases: ["zcsp"],
    rootCommandDescription: "Welcome to the sample plug-in for Zowe CLI!\n\n The sample plug-in " +
        "(& CLI) follows the Zowe CLI command syntax 'zowe [group] [action] [object] [options]'. " +
        "Where, in the case of the plugin, " +
        "the [group] is the package.json name, " +
        "the [actions] are defined in the project under 'src/cli/', " +
        "& the [objects] (definitions & handler) are defined in the project under 'src/cli/'.",
    productDisplayName: "Zowe CLI Sample Plug-in",
    name: "zowe-cli-sample"
};

export = config;
