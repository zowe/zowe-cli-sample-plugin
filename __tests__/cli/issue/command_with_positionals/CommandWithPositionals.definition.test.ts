/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import * as fs from "fs";
import { Imperative } from "@brightside/imperative";

describe("command-with-positionals definition", () => {
    it("should match the snapshot", () => {

        // Attempt to read the full file contents. We could require the module here, however there is normally non
        // deterministic data (filepaths, etc.) that are resolved when the module is loaded, so it is simpler to
        // check the contents for changes (sanity/protection agaisnt undesired changes to the definition)
        let contents: string;
        let error;
        try {
            contents = fs.readFileSync(__dirname +
                "/../../../../src/cli/issue/command_with_positionals/CommandWithPositionals.definition.ts").toString();
        } catch (e) {
            error = e;
            Imperative.console.error(`Error reading CommandWithPositionals.definition.ts Did you move the file? Details: ${e.message}`);
        }
        expect(error).toBeUndefined();
        expect(contents).toMatchSnapshot();
    });
});
