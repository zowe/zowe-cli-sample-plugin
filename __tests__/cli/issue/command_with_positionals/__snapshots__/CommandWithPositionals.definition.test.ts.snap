// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command-with-positionals definition should match the snapshot 1`] = `
"/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ICommandDefinition } from \\"@brightside/imperative\\";
/**
 * Command one [object] defintion. This definition is of imperative type \\"command\\" and therefore must have a
 * command handler (which performs the \\"work\\" for this command).
 *
 * In this case, \\"command-with-positionals\\" is a command that requires positional arguments
 *
 * Property Summary:
 * =================
 * \\"name\\" of the [object]. Should be a noun (e.g. data-set)
 * \\"aliases\\" normally contains a shortened form of the command
 * \\"summary\\" will display when issuing the help on this [objects] [action]
 * \\"type\\" is \\"command\\" which means a handler is required
 * \\"handler\\" is the file path to the handler (does the work)
 * \\"positionals\\" is an array of positional parameters
 */
export const CommandWithPositionalsDefinition: ICommandDefinition = {
    name: \\"command-with-positionals\\",
    aliases: [\\"cwp\\"],
    summary: \\"[object] command-with-positionals prints console msgs\\",
    description: \\"[objects] in Zowe CLI are the entities on which [actions] are perfomed. [objects] are always nouns. \\" +
        \\"For example, for command \\\\\\"zowe zos-files delete data-set\\\\\\", the [object] is \\\\\\"data-set\\\\\\". For this command, \\" +
        \\"we will simply print console error (stderr) messages and the command should succeed.\\",
    type: \\"command\\",
    handler: __dirname + \\"/CommandWithPositionals.handler\\",
    positionals: [
        {
            name: \\"firstpositional\\",
            type: \\"string\\",
            description: \\"The first positional parameter\\",
            required: true
        },
        {
            name: \\"secondpositional\\",
            type: \\"number\\",
            description: \\"The second positional parameter\\",
            required: true
        }
    ]
};
"
`;
